pipeline/dbt/models/intermediate/sources/fredo/int_fredo__services.sql

```diff
@@ -0,0 +1,302 @@
+        ('santé mentale', ARRAY['sante--bien-etre-psychologique']),
+        ('scolarité', ARRAY['accompagnement-social-et-professionnel-personnalise--decrochage-scolaire', 'illettrisme--accompagner-scolarite', 'famille--information-et-accompagnement-des-parents']),
+        ('soins à domicile', ARRAY['sante--acces-aux-soins']),
+        ('sortir de l’isolement', ARRAY['remobilisation--lien-social']),
+        ('soutien enfants déscolarisés / scolarité adaptée', ARRAY['accompagnement-social-et-professionnel-personnalise--decrochage-scolaire']),
+        ('soutien scolaire', ARRAY['famille--information-et-accompagnement-des-parents', 'illettrisme--accompagner-scolarite']),
+        ('sport', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sport de musculation', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sport de réflexion', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports aquatiques', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports cyclistes', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports d''athlétisme', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports de ballon', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports de combat', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports de détente', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports de plein air', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports de raquette', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports équestres', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports gymniques', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports mécaniques', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('sports nautiques', ARRAY['remobilisation--decouvrir-son-potentiel-via-le-sport-et-la-culture']),
+        ('stages', ARRAY['accompagnement-social-et-professionnel-personnalise--definition-du-projet-professionnel', 'choisir-un-metier--decouvrir-un-metier-ou-un-secteur-dactivite']),
+        ('structures d’insertion par l’emploi', ARRAY['accompagnement-social-et-professionnel-personnalise--parcours-d-insertion-socioprofessionnel']),
+        ('transport', ARRAY['mobilite--comprendre-et-utiliser-les-transports-en-commun', 'mobilite--etre-accompagne-dans-son-parcours-mobilite']),
+        ('vêtements', ARRAY['equipement-et-alimentation--habillement']),
+        ('vih', ARRAY['sante--vie-relationnelle-et-affective'])
+    ) AS x (category, thematiques)
+),
```

Comment:
une autre façon de procéder (peut-être un poil plus lisible et sans `LATERAL`, donc moins complexe)

```suggestion
        ('transport', 'mobilite--comprendre-et-utiliser-les-transports-en-commun'),
        ('transport', 'mobilite--etre-accompagne-dans-son-parcours-mobilite'),
        ('vêtements', 'equipement-et-alimentation--habillement'),
        ('vih', 'sante--vie-relationnelle-et-affective')
    ) AS x (category, thematique)
),
```

en gros on duplique des category
et du coup ça donne un simple JOIN+GROUP BY plus bas :

```sql
thematiques AS (
    SELECT
        categories.structure_id,
        ARRAY_AGG(DISTINCT thematique) AS thematiques
    FROM categories
    LEFT JOIN di_thematiques_by_fredo_categories ON categories.value = di_thematiques_by_fredo_categories.category
    GROUP BY categories.structure_id
),
```

Reply:
je pense que c'est mieux sans LATERAL, voir même en lisibilité, mais c'est surement + subjectif

si tu veux, dans le fichier séparé, t'as aussi l'option d'utiliser un format json (cf par exemple ce qui est fait https://github.com/gip-inclusion/data-inclusion/blob/main/pipeline/dbt/models/staging/sources/soliguide/stg_soliguide__categories.sql)

Reply:
Pas sur que le json accete des keys identique au sein d'un meme objet. Je vais rester en sql je pense. Ok pour enlever le lateral :)

Reply:
pardon je n'ai pas été très clair : tu peux formatter en json si tu veux garder le format clé -> array pour la lisibilité du mapping (ou juste avec du SQL bien formatté sans json), du moment que tu gardes le format de sortie du model qui permet de se passer de lateral bien entendu

Reply:
non mais ca me va comme ca (+ un peu la flemme de le faire une troisieme fois il y a quand meme pas mal de lignes)